cocotb/monitors/__init__.py

```diff
@@ -174,14 +174,16 @@ class BusMonitor(Monitor):
     _optional_signals = []
 
     def __init__(self, entity, name, clock, reset=None, reset_n=None,
-                 callback=None, event=None, bus_separator="_", array_idx=None):
+                 callback=None, event=None, bus_separator="_", array_idx=None,
+                 **kwargs):
         self.log = SimLog("cocotb.%s.%s" % (entity._name, name))
         self.entity = entity
         self.name = name
         self.clock = clock
         self.bus = Bus(self.entity, self.name, self._signals,
                        optional_signals=self._optional_signals,
-                       bus_separator=bus_separator, array_idx=array_idx)
+                       bus_separator=bus_separator, array_idx=array_idx,
+                       **kwargs)
```

Comment:
I'm not sold on this - what kwargs are you expecting?

Reply:
In follow-up patch introducing ``TypedBus`` the Bus class will not be fixed and different call signatures need to be supported. But I can do that in the follow-up patch then. I will also plan to rename kwargs to bus_kwargs then. I can either move this patch to follow-up patch or do the following:
```python
class BusMonitor(Monitor):
    """Wrapper providing common functionality for monitoring buses."""
    _signals = []
    _optional_signals = []

    def __init__(self, entity, name, clock, reset=None, reset_n=None,
                 callback=None, event=None, **bus_kwargs):
        self.log = SimLog("cocotb.%s.%s" % (entity._name, name))
        self.entity = entity
        self.name = name
        self.clock = clock
        self.bus = Bus(self.entity, self.name, self._signals,
                       optional_signals=self._optional_signals,
                       **bus_kwargs)
```

Reply:
Let's move this to the follow-up, I think it will be easier to understand in context.

Reply:
I'll move it to follow-up patch, it's more natural there.